skbase/tests/test_baseestimator.py

```diff
@@ -81,3 +82,45 @@ def test_check_is_fitted_raises_error_when_unfitted(fixture_estimator_instance):
 
     fixture_estimator_instance._is_fitted = True
     assert fixture_estimator_instance.check_is_fitted() is None
+
+
+class FittableCompositionDummy(BaseEstimator):
+    """Potentially composite object, for testing."""
+
+    def __init__(self, foo, bar=84):
+        self.foo = foo
+        self.foo_ = deepcopy(foo)
+        self.bar = bar
+
+    def fit(self):
+        """Fit, dummy."""
+        if hasattr(self.foo_, "fit"):
+            self.foo_.fit()
+        self._is_fitted = True
+
+
+def test_get_fitted_params():
```

Comment:
Should the test class and test also look for a simple  fitted param? E.g., in ffit something like some_fitted_param_ gets a value assigned?

Reply:
That's what is being tested in line 121, `assert set(non_comp_f_params.keys()) == {"foo"}`

Or, do you mean sth different? If so, kindly expand.

Reply:
Ah -- I didn't see where `foo` was being assigned. It is assigned to `foo_` attribute in `__init__ ` and not `fit` so I missed it. Seems like your doing that for ease to test composite and non-composite case. It works, and proves that an attribute that ends in "_" but doesn't start with "_" will get returned as a fitted param.

Reply:
so, this is resolved then, right?